src/app/pages/token/token-list/token-cw20/token-cw20.component.ts

```diff
@@ -235,76 +274,127 @@ export class TokenCw20Component implements OnInit, OnDestroy {
-    this.sortedData = data.sort((a, b) => {
-      switch (sort.active) {
-        case 'price':
-          return this.compare(a.price, b.price, isAsc);
-        case 'volume':
-          return this.compare(a.volume, b.volume, isAsc);
-        case 'circulating_market_cap':
-          return this.compare(a.circulating_market_cap, b.circulating_market_cap, isAsc);
-        case 'onChainMarketCap':
-          return this.compare(a.onChainMarketCap, b.onChainMarketCap, isAsc);
-        default:
-          return 0;
-      }
-    });
+  //   const isAsc = sort.direction === 'asc';
+  //   this.sortedData = data.sort((a, b) => {
+  //     switch (sort.active) {
+  //       case 'price':
+  //         return this.compare(+a.price, +b.price, !isAsc);
+  //       case 'totalSupply':
+  //         return this.compare(+a.totalSupply, +b.totalSupply, !isAsc);
+  //       case 'inChainValue':
+  //         return this.compare(+a.inChainValue, +b.inChainValue, !isAsc);
+  //       default:
+  //         return 0;
+  //     }
+  //   });
 
-    if (sort.active === 'change') {
-      let lstUp = this.sortedData
-        .filter((data) => data.isValueUp)
-        ?.sort((a, b) => this.compare(a.change, b.change, isAsc));
-      let lstDown = this.sortedData
-        .filter((data) => !data.isValueUp)
-        .sort((a, b) => this.compare(a.change, b.change, !isAsc));
-      this.sortedData = isAsc ? lstDown.concat(lstUp) : lstUp.concat(lstDown);
-    }
-    const payload = {
-      limit: this.pageData.pageSize,
-      offset: this.pageData.pageIndex * this.pageData.pageSize,
-    };
+  //   if (sort.active === 'change') {
+  //     let lstUp = this.sortedData
+  //       .filter((data) => data.isValueUp)
+  //       ?.sort((a, b) => this.compare(a.change, b.change, isAsc));
+  //     let lstDown = this.sortedData
+  //       .filter((data) => !data.isValueUp)
+  //       .sort((a, b) => this.compare(a.change, b.change, !isAsc));
+  //     this.sortedData = isAsc ? lstDown.concat(lstUp) : lstUp.concat(lstDown);
+  //   }
+  //   const payload = {
+  //     limit: this.pageData.pageSize,
+  //     offset: this.pageData.pageIndex * this.pageData.pageSize,
+  //   };
 
-    let dataFilter = this.sortedData.slice(payload?.offset, payload?.offset + payload?.limit);
-    this.pageData = {
-      length: this.pageData.length,
-      pageSize: this.pageData.pageSize,
-      pageIndex: this.pageData.pageIndex,
-    };
-    this.dataSource.data = dataFilter;
-  }
+  //   let dataFilter = this.sortedData.slice(payload?.offset, payload?.offset + payload?.limit);
+  //   this.pageData = {
+  //     length: this.pageData.length,
+  //     pageSize: this.pageData.pageSize,
+  //     pageIndex: this.pageData.pageIndex,
+  //   };
+  //   this.dataSource.data = dataFilter;
+  // }
 
   compare(a: number | string, b: number | string, isAsc: boolean) {
-    return (a < b ? -1 : 1) * (isAsc ? 1 : -1);
+    return (BigNumber(a).lt(b) ? -1 : 1) * (isAsc ? 1 : -1);
```

Comment:
Chuyển sang dùng Bignumber
```suggestion
    return (BigNumber(a).lt(BigNumber(b)) ? -1 : 1) * (isAsc ? 1 : -1);
```